web_src/js/components/DashboardRepoList.js

```diff
@@ -301,29 +301,23 @@ function initVueComponents() {
       searchRepos() {
         this.isLoading = true;
 
-        if (!this.reposTotalCount) {
-          const totalCountSearchURL = `${this.subUrl}/api/v1/repos/search?sort=updated&order=desc&uid=${this.uid}&team_id=${this.teamId}&q=&page=1&mode=`;
-          $.getJSON(totalCountSearchURL, (_result, _textStatus, request) => {
-            this.reposTotalCount = request.getResponseHeader('X-Total-Count');
-          });
-        }
-
         const searchedMode = this.repoTypes[this.reposFilter].searchMode;
         const searchedURL = this.searchURL;
         const searchedQuery = this.searchQuery;
 
-        $.getJSON(searchedURL, (result, _textStatus, request) => {
+        fetch(searchedURL).then(async (response) => {
+          const json = await response.json();
```

Comment:
Had to use `await here` as you both the response and json variable here, could be decoupled into 2x then, but that's IMO unnecessary complexity

Reply:
If you want to use `fetch`, the code usually looks like this:

```js
try {
    loading = true
    const resp = await fetch()
    const data = await resp.json()
    ....
} finally {
    loading = false
}
```

Reply:
I prefer it slightly differently:

```js
this.isLoading = true;

let data;
try {
  const res = await fetch()
  data = await resp.json()
} catch {
  return this.isLoading = false;
}

// use data
```

Reply:
`catch` above will catch all errors, there's no need for `finally` which is just syntactic sugar, I never use it. Just need to set `isLoading` after data processing is done, edited that in.

Reply:
The case in my demo is not the same as this PR. With this PR, `isLoading` doesn't control the logic, so your solution is also fine.

However, for cases that `isLoading` controls the logic (eg: preventing duplicated requests), a `finally` block covers all code is generally more safe. And yes most JS features are impelemented by sugar syntax ..... it exists because it has its values.

Reply:
https://github.com/go-gitea/gitea/pull/18963/commits/22acfd62f9ebc10ecbffb01a46504c86d9dcd4ab